apolo-sdk/src/apolo_sdk/_apps.py

```diff
@@ -0,0 +1,72 @@
+from dataclasses import dataclass
+from typing import AsyncIterator, Optional
+
+from ._config import Config
+from ._core import _Core
+from ._rewrite import rewrite_module
+from ._utils import NoPublicConstructor, asyncgeneratorcontextmanager
+
+
+@rewrite_module
+@dataclass(frozen=True)
+class App:
+    id: str
+    name: str
+    display_name: str
+    template_name: str
+    template_version: str
+    project_name: str
+    org_name: str
+    state: str
+
+
+@rewrite_module
+class Apps(metaclass=NoPublicConstructor):
+    def __init__(self, core: _Core, config: Config) -> None:
+        self._core = core
+        self._config = config
+
+    @asyncgeneratorcontextmanager
+    async def list(
+        self,
+        cluster_name: Optional[str] = None,
+        org_name: Optional[str] = None,
+        project_name: Optional[str] = None,
+    ) -> AsyncIterator[App]:
+        cluster_name = cluster_name or self._config.cluster_name
+        if org_name is None:
+            org_name = self._config.org_name
+            if org_name is None:
+                raise ValueError("Organization name is required")
+        if project_name is None:
+            project_name = self._config.project_name
+            if project_name is None:
+                raise ValueError("Project name is required")
```

Comment:
If I directly pass, say, `org_name`, I cannot use default `project_name`, isn't it?
Please carefully process all possible argument combinations where defaults are applicable and where they are not.